### ISSUE TITLE
Default Dark Theme highlight issue

### ISSUE THREAD

**Issue by vvatikiotis** [Microsoft/vscode#16145]
- VSCode Version: 1.7.2
- OS Version: 10.11.6

Applying the default Dark Theme on a piece of javascript code:
<img width="835" alt="untitled" src="https://cloud.githubusercontent.com/assets/535853/20669314/350eae1c-b57c-11e6-8945-7206bd76209c.png">

This is clearly a bug. String highlight colour leaks past the string, to the rest of the code.

**Comment by mjbvz** [comment]
@vvatikiotis can you please paste the code up here so that we can test this. We don't need the whole file, only the shortest possible snipped that shows this behavior.

 Thanks.

**Comment by vvatikiotis** [comment]
Here you are. It exhibits the behaviour you see in the screenshot.

```
const generateLookupReducerFor = (constants, action, key, loadingKey, selectedKey) => ({
  [constants[`FETCH_${action}_PENDING`]]: (state, { payload }) => {
    return mergeProp('filters', state, { [loadingKey]: true })
  },
  [constants[`FETCH_${action}_FAILURE`]]: (state, { payload }) => {
    return mergeProp('filters', state, { [loadingKey]: false })
  },
  [constants[`FETCH_${action}_SUCCESS`]]: (state, { payload }) => {
    return mergeProp('filters', state, {
      [loadingKey]: false,
      [key]: payload,
    })
  },
})
```

**Comment by mjbvz** [comment]
@vvatikiotis Thanks.  I believe this is fixed in the latest insiders builds:

![screen shot 2016-11-29 at 10 09 24 am](https://cloud.githubusercontent.com/assets/12821956/20722613/1ccd9c82-b61c-11e6-9b2b-2dd3d465517a.png)

1.7.2 is below, with 1.8 insiders shown above.

Please give the [1.8 insiders builds](https://code.visualstudio.com/insiders) a try and let me know if you still see this issue.

**Comment by vvatikiotis** [comment]
Looks fixed in 1.8 insiders build. Thanks!
